### ISSUE TITLE
Psalm & PHPStorm Plugin integration

### ISSUE THREAD

**Issue by mathroc** [vimeo/psalm#7091]
PHPStorm plugin for Psalm create a temporary file outside of the project and calls Psalm on it to get validation in the IDE without having to save the file in the project repository. 

This causes issues with how Psalm handle path to ignore files and to suppress issues. More can be found there : https://youtrack.jetbrains.com/issue/WI-59061

I don't know if there's something to do on Psalm side or not. But maybe you have some suggestion for the PHPStorm plugin

**Comment by orklah** [comment]
I'm not sure I get what the issue really is.

Their setup seems to work for most people, so I'd guess this is related to a specific config. They're talking about `resolveFromConfigFile` at one point saying that putting this to false could be a solution but `resolveFromConfigFile` default is true, so I'd expect a lot more people to have this issue.

I'm also not sure what change they could be making to the config file (given they relocate some file, it could make sense they also edit a temporary config file too...)

Do you have this issue on your project? Are you able to reproduce on a small repo?

**Comment by mathroc** [comment]
I'm not using PHPStorm myself, I'll ask one of my coworker if he can do that

we tried changing `resolveFromConfigFile` to `false` but this did not help.

Later, I realized that I'm using Psalm with VSCode and it's working great. But I notice one major difference between the 2 plugins, VSCode uses `psalm --language-server` but PHPStorm runs it manually for every file they want to check. The performance is not so great: even with cache enabled, running `psalm -c ... /tmp-path-to/file.php` (like PHPStorm plugin is doing) takes about 3 seconds on our project. 
So we tried [another plugin that brings LSP support to PHPStorm](https://plugins.jetbrains.com/plugin/10209-lsp-support) and configured it for Psalm and this is working just fine (and much faster)
